code/game/machinery/kitchen/cooking_machines/_cooker.dm

```diff
@@ -1,236 +1,145 @@
-	return ..()
-
-/obj/machinery/cooker/examine()
-	..()
-	if(cooking_obj && Adjacent(usr))
-		usr << "You can see \a [cooking_obj] inside."
-
-/obj/machinery/cooker/attackby(var/obj/item/I, var/mob/user)
-	if(!cook_type || (stat & (NOPOWER|BROKEN)))
-		user << "<span class='warning'>\The [src] is not working.</span>"
-		return
-
-	if(cooking)
-		user << "<span class='warning'>\The [src] is running!</span>"
-		return
-
-	if(!dropsafety(I))
-		return
-
-	// We are trying to cook a grabbed mob.
-	var/obj/item/weapon/grab/G = I
-	if(istype(G))
-
-		if(!can_cook_mobs)
-			user << "<span class='warning'>That's not going to fit.</span>"
-			return
-
-		if(!isliving(G.affecting))
-			user << "<span class='warning'>You can't cook that.</span>"
-			return
-
-		cook_mob(G.affecting, user)
-		return
-
-	// We're trying to cook something else. Check if it's valid.
-	var/obj/item/weapon/reagent_containers/food/snacks/check = I
-	if(istype(check) && islist(check.cooked) && (cook_type in check.cooked))
-		user << "<span class='warning'>\The [check] has already been [cook_type].</span>"
-		return 0
-	else if(istype(check, /obj/item/weapon/reagent_containers/glass))
-		user << "<span class='warning'>That would probably break [src].</span>"
-		return 0
-	else if(istype(check, /obj/item/weapon/disk/nuclear))
-		user << "Central Command would kill you if you [cook_type] that."
-		return 0
-	else if(!istype(check) && !istype(check, /obj/item/weapon/holder))
-		user << "<span class='warning'>That's not edible.</span>"
-		return 0
-
-	// Gotta hurt.
-	if(istype(cooking_obj, /obj/item/weapon/holder))
-		for(var/mob/living/M in cooking_obj.contents)
-			M.apply_damage(rand(30,40), BURN, "chest")
-
-	if(!user.unEquip(I))
-		return
-
-	// We can actually start cooking now.
-	user.visible_message("<span class='notice'>\The [user] puts \the [I] into \the [src].</span>")
-	cooking_obj = I
-	cooking_obj.forceMove(src)
-	cooking = 1
-	icon_state = on_icon
-
-	sleep(cook_time)
-
-	// Sanity checks.
-	if (!check_cooking_obj())
-		return
-
-	if(istype(cooking_obj, /obj/item/weapon/holder))
-		for(var/mob/living/M in cooking_obj.contents)
-			M.death()
-
-	// Cook the food.
-	var/cook_path
-	if(selected_option && output_options.len)
-		cook_path = output_options[selected_option]
-	if(!cook_path)
-		cook_path = /obj/item/weapon/reagent_containers/food/snacks/variable
-	var/obj/item/weapon/reagent_containers/food/snacks/result = new cook_path(src) //Holy typepaths, Batman.
-
-	if(cooking_obj && cooking_obj.reagents && cooking_obj.reagents.total_volume)
-		cooking_obj.reagents.trans_to(result, cooking_obj.reagents.total_volume)
-
-	// Set icon and appearance.
-	change_product_appearance(result)
-
-	// Update strings.
-	change_product_strings(result)
-
-	// Set cooked data.
-	var/obj/item/weapon/reagent_containers/food/snacks/food_item = cooking_obj
-	if(istype(food_item) && islist(food_item.cooked))
-		result.cooked = food_item.cooked.Copy()
+/obj/machinery/appliance/cooker
```

Comment:
Your heating code seems quite complicated.
When you can achieve a minimum temperature with something as simple as a bi-metal switch that turns off as soon as a certain minimum temperature is reached.
(Full power until the min temp; Often used in smaller melting ovens, tea water heaters and simple ovens).